torch/_inductor/decomposition.py

```diff
@@ -647,3 +647,19 @@ def masked_scatter(self, mask, source):
         source_idx = mask.reshape(-1).cumsum(0) - 1
         return inductor_prims.masked_scatter_with_index(self, mask, source_idx, source)
     return NotImplemented
+
+
+@register_decomposition(aten.put)
+def put(self, index, source, accumulate=False):
+    stride = utils.compute_elementwise_output_strides(self)
+    flattened = self.flatten()
+    flattened = torch.index_put(
+        flattened, [index], source.reshape(index.shape), accumulate
+    )
+    return flattened.reshape(self.shape)
+
+
+@register_decomposition(aten.put_)
+def put_(self, index, source, accumulate=False):
+    out = aten.put(self, index, source, accumulate=accumulate)
+    return self.copy_(out)
```

Comment:
Not needed anymore

Reply:
Not sure about this, see https://github.com/pytorch/pytorch/pull/120179#discussion_r1494800076